training/src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -33,22 +35,34 @@
     @Autowired
     private BookRepository bookRepository;
 
+    @Autowired
+    private SecurityService securityService;
+
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    public User create(@RequestBody User user) {
+    public User create(@RequestBody User user) throws UsernameExistsException {
+        if (userRepository.findFirstByUsername(user.getUsername()).isPresent())
+            throw new UsernameExistsException(user.getUsername() + "already exists");
         return userRepository.save(user);
     }
 
     @PutMapping("/{id}")
-    public User updateUser(@RequestBody User user, @PathVariable int id) {
-        if (user.getId() != id) {
+    public User updateUser(@RequestBody UserDTO userDTO, @PathVariable int id) {
+        if (userDTO.getId() != id) {
             throw new UserIdMismatchException();
         }
-        userRepository.findById(id)
+        User prevUser = userRepository.findById(id)
             .orElseThrow(UserNotFoundException::new);
+        User user = new User(userDTO);
+        user.setPassword(prevUser.getPassword());
```

Comment:
In this case, I think is better to have a method like "setFromDTO" in User, instead of using a constructor. Because is the same object, I believe that you don't need to instantiate a new one, right? So you won't even need to setPassword because the object will remain the same